example_client/grpc_stateful_client.py

```diff
@@ -26,160 +26,356 @@
+    parser.add_argument(
+        '--sequence_id',
+        required=False,
+        default=1,
+        help='Starting unique sequence id. Default: 1')
+
+    print('### Starting grpc_stateful_client.py client ###')
+
+    args = vars(parser.parse_args())
+    return args
+
+
+def prepare_processing_data(args):
+    delimiter = ","
+    input_files = []
+    reference_files = []
+    input_names = []
+    output_names = []
+    input_paths = args.get('input_path').split(delimiter)
+    for input_path in input_paths:
+        print('Reading input ark file {}'.format(input_path))
+        ark_reader = ArchiveReader(input_path)
+        input_files.append(ark_reader)
+
+    score_paths = args.get('score_path').split(delimiter)
+    for score_path in score_paths:
+        print('Reading scores ark file {}'.format(score_path))
+        ark_score = ArchiveReader(score_path)
+        reference_files.append(ark_score)
+
+    inputs = args['input_name'].split(delimiter)
+    for input in inputs:
+        print('Adding input name {}'.format(input))
+        input_names.append(input)
+
+    outputs = args['output_name'].split(delimiter)
+    for output in outputs:
+        print('Adding output name {}'.format(output))
+        output_names.append(output)
+
+    # Validate input
+    if len(input_files) != len(input_names):
+        print(
+            "ERROR: Number of input ark files {} must be equal to the number of input names {}".format(
+                len(input_files),
+                len(input_names)))
+        exit(1)
+
+    if len(reference_files) != len(output_names):
+        print(
+            "ERROR: Number of output ark files {} must be equal to the number of output names {}".format(
+                len(input_files),
+                len(input_names)))
+        exit(1)
+
+    # Consolidate input
+    input_data = dict()
+    name_index = 0
+    for ark_reader in input_files:
+        input_name = input_names[name_index]
+        input_sub_data = dict()
+        for name_key, data_obj in ark_reader:
+            sequence_size = data_obj.shape[0]
+            data = list()
+            for input_index in range(0, sequence_size):
+                data.append(np.expand_dims(data_obj[input_index], axis=0))
+            input_sub_data[name_key] = data
+        input_data[input_name] = input_sub_data
+        name_index += 1
+
+    # Consolidate reference scores
+    name_index = 0
+    reference_scores = dict()
+    for ark_score in reference_files:
+        output_name = output_names[name_index]
+        score_ojects = {k: m for k, m in ark_score}
+        reference_scores[output_name] = score_ojects
+        name_index += 1
+
+    # First ark file is the one we will iterate through for all input ark
+    # files per inference request input
+    data_iterator = dict()
+    for key, obj in input_files[0]:
+        data_iterator[key] = obj
```

Comment:
```suggestion
    data_iterator = dict()
    for key, obj in input_files[0]:
        data_iterator[key] = obj
```
```suggestion
    sequence_size_map= dict()
    for sequence_name, data in input_files[0]:
        sequence_size_map[sequence_name] = data.shape[0]
```
With that change below code might be more readable.